### DISCUSSION TITLE
How to properly run some code after validation and before the relationship is saved in Admin Panel?

### DISCUSSION THREAD

**Discussion by hizzely** [filamentphp/filament#2736] [Help] [answered]
First and foremost, thanks for this great package! :)

So I have a column (not in the form) that needs to be updated whenever a specific foreign key is changed. 

**Here's what I've tried:**
I can't do this in `handleRecordUpdate()` since I don't have the old FK value to compare because it was updated first when `getState(shouldCallHooksBefore: true)` function is called in the beginning of the `save()` function.

I solved the problem by overriding the `save()` function instead, just like in the [documentation](https://filamentphp.com/docs/2.x/forms/getting-started#field-relationships). However, now I have to copy the entire `save()` code to my `EditResource` file to make the admin panel saving process works again.

**My question is:**
Is this the correct way to do it in Filament? I'm aware that listening for Eloquent's `updating` event is also a possible option. But to make things simpler (supposedly), I decided to go with that option as a last resort.

Thanks!

**Comment by danharrin** [comment, marked as answer]
Regardless of how I would approach this in Filament, I suggest that you use a Model observer (updating event), since this sounds like a situation where you'd also want this logic outside the admin panel. It's less a concern of the admin panel, more a concern of the model layer.

**Reply by hizzely** [reply]
Thanks for the suggestion! Yes I agree that Model Observer might be the way to go. But currently only the admins are allowed to change the FK, so it only applies in admin panel. That's why I think putting the logic in admin panel is the simpler option and wondering if there's a way to do this in Filament.

I think I'm going to use Model Observer instead, just like you've suggested. Because overriding the `save()` function and duplicating the code also doesn't feel right to me.
